### PR TITLE
Fix datasize call in calibrate

### PR DESCRIPTION
**Pull request by vuvko**
**Description**: Created an abstract property `datasize` for the `CalibrationDataReader` class.

**Motivation and Context**
- fixes https://github.com/microsoft/onnxruntime/issues/5109

### CODE REVIEW

File: onnxruntime/python/tools/quantization/calibrate.py [hunk 1]
```diff
@@ -23,6 +23,12 @@ def __subclasshook__(cls, subclass):
     def get_next(self) -> dict:
         """generate the input data dict for ONNXinferenceSession run"""
         raise NotImplementedError
+    
+    @property
+    @abc.abstractmethod
+    def datasize(self) -> int:
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/calibrate.py:29]: there is no need to add this property. Could you please change:
            dict(zip(node_output_names, intermediate_outputs[i])) for i in range(self.data_reader.datasize) 
to:
            dict(zip(node_output_names, intermediate_output)) for intermediate_output in intermediate_outputs

Comment (vuvko) [reply onnxruntime/python/tools/quantization/calibrate.py:29]: Oh, OK, didn't realize this easier solution.

**Comment by vuvko** [comment]
@yufenglee comment addressed, I ran the e2e example and everything worked.

**Review by yufenglee — APPROVED** [review]
Thank your very much for your contribution!

**Pull request merged**